### ISSUE TITLE
Evidence uploader API documentation

### ISSUE THREAD

**Issue by solomondefi-dev** [SolomonDefi/solomon-monorepo#67]
Review FastAPI's suggested strategy for documenting API endpoints, and implement the necessary metadata: https://fastapi.tiangolo.com/tutorial/metadata/

- Add documentation to existing endpoints
- See if there's an easy way to export HTML for use in the `developer-docs` app

**Comment by kelvin-wong** [comment]
@solomondefi-dev 
The way FastAPI to generating the documentation is by serving a static HTML with SwaggerUI js to generate the API doc with `openapi.json` as below. So the simplest way to integrate API docs, we just need to create the SwaggerUI js app on a static page.

```html
<div id="swagger-ui">
</div>
<script src="https://cdn.jsdelivr.net/npm/swagger-ui-dist@3/swagger-ui-bundle.js"></script>
<script>
const ui = SwaggerUIBundle({
    url: '/api/openapi.json',
    oauth2RedirectUrl: window.location.origin + '/docs/oauth2-redirect',
    dom_id: '#swagger-ui',
    presets: [
        SwaggerUIBundle.presets.apis,
        SwaggerUIBundle.SwaggerUIStandalonePreset
    ],
    layout: "BaseLayout",
    deepLinking: true,
    showExtensions: true,
    showCommonExtensions: true
})
</script>
```

**Comment by solomondefi-dev** [comment]
Ok great. I will change this to a docs issue and handle the integration in that app.

What's the process for generating `openapi.json`? Can we automate that somehow?

**Comment by kelvin-wong** [comment]
> Ok great. I will change this to a docs issue and handle the integration in that app.
> 
> What's the process for generating `openapi.json`? Can we automate that somehow?

`openapi.json` will be generated by FastAPI according to the routes, model schemas and responses. That's why we only need to a static page link back to the `openapi.json`.

**Comment by solomondefi-dev** [comment]
Right, but when exactly is `openapi.json` generated? Whenever the dev server is run?

**Comment by kelvin-wong** [comment]
> Right, but when exactly is `openapi.json` generated? Whenever the dev server is run?

`openapi.json` is served by the server and generated on request.

**Comment by solomondefi-dev** [comment]
I see, so the `api-evidence` app needs to be running in that situation. Ideally we'd generate `openapi.json` and bundle it with the developer docs app on deploy.

For it looks like this is possible: https://github.com/tiangolo/fastapi/issues/1173

```python
from fastapi.openapi.utils import get_openapi
from main import app
import json


with open('openapi.json', 'w') as f:
    json.dump(get_openapi(
        title=app.title,
        version=app.version,
        openapi_version=app.openapi_version,
        description=app.description,
        routes=app.routes,
        openapi_prefix=app.openapi_prefix,
    ), f)
```

So in the developer docs release CI, we can call that script and bundle the resulting `openapi.json` with the docs app.

**Comment by solomondefi-dev** [comment]
I'm not sure why this wasn't closed in connection with the PR, since the issue is linked 😅
